File: scripts-ruby/commands/parse_data.rb
require_relative '../qtools/qfil'
require_relative '../qtools/qcli'
require_relative '../qtools/qstr'

lines = QFil.get_lines_from_file("data/flashcards.txt")
QCli.message("Read #{lines.size} lines from flashcards.txt")

flashcards = []
i = 0
while i < lines.size
	suuid = QStr.generate_suuid
	category = lines[i]&.strip || ""
	front = lines[i + 1]&.strip || ""
	back = lines[i + 2]&.strip || ""

	if !category.empty? && !front.empty? && !back.empty?
		flashcards << {
			'suuid' => suuid,
			'category' => category,
			'front' => front,
			'back' => back
		}
	else
		QCli.message(
			"Skipping incomplete flashcard at lines #{i + 1}-#{i + 3}",
			"warning"
		)
	end
	i += 4
end

QCli.message("Generated #{flashcards.size} flashcards")

# Save flashcards to JSON file
require 'json'
json_data = JSON.pretty_generate(flashcards)
begin
	QFil.write_text_block_to_file("parseddata/flashcards.json", json_data)
	QCli.message("Wrote #{flashcards.size} flashcards to flashcards.json")
rescue
	QCli.message("Error writing flashcards to JSON file", "error")
end

File: scripts/parse-data.py
import json

from qtools import *

lines = qfil.get_lines_from_file("../../../../webs/italian/texts/itnotes.html")
verbs = []

for line in lines:
	if "//" in line:
		verb = line
		if ":" in verb:
			verb = verb.split(":", 1)[0]
			verb = verb.split("#", 1)[0]
			verb = verb.split("//", 1)[0]
			verb = verb.strip()
			verbs.append(verb)

print(f"{len(verbs)} verbs")
# for verb in verbs:
# 	print(f"\"{verb}\"")

try:
	json_data = json.dumps(verbs, indent=4)
	
	# Write JSON data to file
	with open("../parseddata/verbs.json", 'w') as json_file:
		json_file.write(json_data)
	
	qcli.message("Successfully updated verbs.json")

except Exception as err:
	qcli.message(f"Error: {err}", "error")
